v2/api/cognitiveservices/customizations/account_extension.go

```diff
@@ -135,15 +120,25 @@ func secretsToWrite(obj *storage.Account, keys map[string]string) ([]*v1.Secret,
 	collector := secrets.NewCollector(obj.Namespace)
 	collector.AddValue(operatorSpecSecrets.Key1, keys[accountKey1])
 	collector.AddValue(operatorSpecSecrets.Key2, keys[accountKey2])
-	collector.AddValue(operatorSpecSecrets.Endpoint, keys[accountEp])
-	collector.AddValue(operatorSpecSecrets.Endpoints, keys[accountEps])
+
+	for k, v := range keys {
+		if k == accountKey1 || k == accountKey2 {
+			continue
+		}
+		collector.AddValue(&genruntime.SecretDestination{
+			Name: operatorSpecSecrets.Key1.Name,
```

Comment:
This isn't correct: If we support endpoints as a value output, users should opt-in to it and have the ability to specify its destination (like we do for CEL), rather than assuming that it's always the location of key1.

Take a look at my comment further down though, I think I have a way to use CEL to do this (curious as to your thoughts on usage), which would allow you to remove this bit entirely from here.